hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Server.java

```diff
@@ -636,10 +637,17 @@ void updateMetrics(Call call, long startTime, boolean connDropped) {
     processingTime -= waitTime;
     String name = call.getDetailedMetricsName();
     rpcDetailedMetrics.addProcessingTime(name, processingTime);
+    // Overall processing time is from arrival to completion.
+    long overallProcessingTime = rpcMetrics.getMetricsTimeUnit()
```

Comment:
Add unit

Reply:
Hi @goiri,

I don't see time unit for other _Times_ in the same function. Should we follow the same pattern or should we add unit for them all? 

```
    long queueTime = details.get(Timing.QUEUE, RpcMetrics.TIMEUNIT);
    rpcMetrics.addRpcQueueTime(queueTime);
    long processingTime =
        details.get(Timing.PROCESSING, RpcMetrics.TIMEUNIT);
    long waitTime =
        details.get(Timing.LOCKWAIT, RpcMetrics.TIMEUNIT);
    rpcMetrics.addRpcLockWaitTime(waitTime);
    rpcMetrics.addRpcProcessingTime(processingTime);
```

Reply:
The problem is the timeUnit can be changed. So, it is a bad idea to hard code them as variable names.

Reply:
I just want to make sure that things that are always one unit are defined as such.
Actually in this call, it looks like the right name would be: `overallProcessingTimeUnit`

Reply:
That sounds like a very odd name to me. Maybe `overallProcessingTimeMetricTimeUnit` or `overallProcessingTimeRpcMetricTimeUnit`? too long?

Reply:
the variable is used immediately at the following line and it is not used anywhere else. I'd argue that it is fine to leave the TimeUnit out. It is clear from the context that it has been converted to the correct timeUnit. Besides, we have other four variables in this function which don't have timeUnit appended to its variable names and we are just following the same convention here.

@goiri, can we move forward with PR? 

```
    long overallProcessingTime = rpcMetrics.getMetricsTimeUnit()
        .convert(completionTimeNanos - arrivalTimeNanos, TimeUnit.NANOSECONDS);
    rpcDetailedMetrics.addOverallProcessingTime(name, overallProcessingTime);
```

Reply:
I have to say it is a little confusing.
So overallProcessingTime will always be nanoseconds right?

Reply:
completionTimeNanos - arrivalTimeNanos will always give us a time duration in nanos.

However, with the following, overallProcessingTime will be converted from nanos to whatever timeUnit is configured for rpcMetrics. Right now, it is converted into mills. But it can be others based on configuration. 
```
    long overallProcessingTime = rpcMetrics.getMetricsTimeUnit()
        .convert(completionTimeNanos - arrivalTimeNanos, TimeUnit.NANOSECONDS);
```

Reply:
mark it as resolved now, as we couldn't come up with a cleaner approach than existing one.